saleor/graphql/product/types.py

```diff
@@ -309,9 +320,23 @@ class Meta:
     def resolve_thumbnail_url(self, info, *, size=None):
         if not size:
             size = 255
-        url = get_thumbnail(self.get_first_image(), size, method='thumbnail')
+        url = get_product_image_thumbnail(
+            self.get_first_image(), size, method='thumbnail')
         return info.context.build_absolute_uri(url)
 
+    def resolve_thumbnail(self, info, *, size=None):
+        if not size:
+            size = 255
+        product_image = self.get_first_image()
+        if product_image:
+            url = get_product_image_thumbnail(
+                product_image, size, method='thumbnail')
+            thumbnail_image = models.ProductImage(
```

Comment:
Is creating a model instance necessary? I think it would be enough if  we did:
```
return ProductImage(
    url=url, alt=product_image.alt, product=product_image.product,
    sort_order=product_image.sort_order)
```

Reply:
For this solution we need to remove `'image', 'product'` from `ProductImage` exclude fields and provide this return:
```
return ProductImage(
                alt=product_image.alt, product=product_image.product,
                image=Image(url), sort_order=product_image.sort_order)
```